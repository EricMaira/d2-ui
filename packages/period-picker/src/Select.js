import React from 'react';
import PropTypes from 'prop-types';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MUISelect from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
    formControl: {
        margin: theme.spacing.unit,
        width: '100%',
    },
});

const Select = ({ name, label, value, onChange, options, classes }) => (
    <FormControl className={classes.formControl}>
        <InputLabel htmlFor={name}>{label}</InputLabel>
        <MUISelect
            value={value}
            onChange={onChange}
            autoWidth={true}
            inputProps={{
                name,
                id: name,
            }}
        >
            {Object.keys(options).map(key => (
                <MenuItem key={key} value={key}>
                    {options[key]}
                </MenuItem>
            ))}
        </MUISelect>
    </FormControl>
);

Select.propTypes = {
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    options: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Select);
