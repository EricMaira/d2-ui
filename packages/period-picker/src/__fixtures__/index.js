export const periodTypes = {
    Daily: 'Daily',
    Weekly: 'Weekly',
    WeeklyWednesday: 'Weekly Wednesday',
    WeeklyThursday: 'Weekly Thursday',
    WeeklySaturday: 'Weekly Saturday',
    WeeklySunday: 'Weekly Sunday',
    BiWeekly: 'Bi weekly',
    Monthly: 'Monthly',
    BiMonthly: 'Bi-monthly',
    Quarterly: 'Quarterly',
    SixMonthly: 'Six monthly',
    SixMonthlyApril: 'Six monthly starting in April',
    SixMonthlyNov: 'Six monthly starting in November',
    Yearly: 'Yearly',
    FinancialApril: 'Financial year starting in April',
    FinancialJuly: 'Financial year starting in July',
    FinancialOct: 'Financial year starting in October',
    FinancialNov: 'Financial year starting in November',
};

export const getPeriodTypesResponse = {
    periodTypes: [
        {
            frequencyOrder: 1,
            name: 'Daily',
            isoDuration: 'P1D',
            isoFormat: 'yyyyMMdd',
        },
        {
            frequencyOrder: 7,
            name: 'Weekly',
            isoDuration: 'P7D',
            isoFormat: 'yyyyWn',
        },
        {
            frequencyOrder: 7,
            name: 'WeeklyWednesday',
            isoDuration: 'P7D',
            isoFormat: 'yyyyWedWn',
        },
        {
            frequencyOrder: 7,
            name: 'WeeklyThursday',
            isoDuration: 'P7D',
            isoFormat: 'yyyyThuWn',
        },
        {
            frequencyOrder: 7,
            name: 'WeeklySaturday',
            isoDuration: 'P7D',
            isoFormat: 'yyyySatWn',
        },
        {
            frequencyOrder: 7,
            name: 'WeeklySunday',
            isoDuration: 'P7D',
            isoFormat: 'yyyySunWn',
        },
        {
            frequencyOrder: 14,
            name: 'BiWeekly',
            isoDuration: 'P14D',
            isoFormat: 'yyyyBiWn',
        },
        {
            frequencyOrder: 30,
            name: 'Monthly',
            isoDuration: 'P1M',
            isoFormat: 'yyyyMM',
        },
        {
            frequencyOrder: 61,
            name: 'BiMonthly',
            isoDuration: 'P2M',
            isoFormat: 'yyyyMMB',
        },
        {
            frequencyOrder: 91,
            name: 'Quarterly',
            isoDuration: 'P3M',
            isoFormat: 'yyyyQn',
        },
        {
            frequencyOrder: 182,
            name: 'SixMonthly',
            isoDuration: 'P6M',
            isoFormat: 'yyyySn',
        },
        {
            frequencyOrder: 182,
            name: 'SixMonthlyApril',
            isoDuration: 'P6M',
            isoFormat: 'yyyyAprilSn',
        },
        {
            frequencyOrder: 182,
            name: 'SixMonthlyNov',
            isoDuration: 'P6M',
            isoFormat: 'yyyyNovSn',
        },
        {
            frequencyOrder: 365,
            name: 'Yearly',
            isoDuration: 'P1Y',
            isoFormat: 'yyyy',
        },
        {
            frequencyOrder: 365,
            name: 'FinancialApril',
            isoDuration: 'P1Y',
            isoFormat: 'yyyyApril',
        },
        {
            frequencyOrder: 365,
            name: 'FinancialJuly',
            isoDuration: 'P1Y',
            isoFormat: 'yyyyJuly',
        },
        {
            frequencyOrder: 365,
            name: 'FinancialOct',
            isoDuration: 'P1Y',
            isoFormat: 'yyyyOct',
        },
        {
            frequencyOrder: 365,
            name: 'FinancialNov',
            isoDuration: 'P1Y',
            isoFormat: 'yyyyNov',
        },
    ],
};
