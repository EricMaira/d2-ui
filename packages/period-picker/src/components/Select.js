import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import MUISelect from '@material-ui/core/Select';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';

const styles = theme => ({
    formControl: {
        margin: theme.spacing.unit,
        width: '100%',
    },
});

export const Select = ({ name, label, value, onChange, options, classes }) => (
    <FormControl className={classes.formControl}>
        <InputLabel htmlFor={name}>{label}</InputLabel>
        <MUISelect
            value={value}
            onChange={onChange}
            autoWidth={true}
            inputProps={{
                name,
                id: name,
            }}
        >
            {Object.keys(options).map(key => (
                <MenuItem key={key} value={key}>
                    {options[key]}
                </MenuItem>
            ))}
        </MUISelect>
    </FormControl>
);

Select.propTypes = {
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    options: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Select);
