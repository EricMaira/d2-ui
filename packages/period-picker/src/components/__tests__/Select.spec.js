import React from 'react';
import { shallow } from 'enzyme';
import { Select } from '../Select';
import { periodTypes } from '../../__fixtures__';

describe('<Select/>', () => {
    const defaultProps = {
        name: 'test',
        label: 'This is a test',
        value: 'Daily',
        onChange: jest.fn(),
        options: periodTypes,
        classes: {
            formControl: 'formControl',
        },
    };
    it('Matches the snapshot for regular fields', () => {
        const wrapper = shallow(<Select {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('Matches the snapshot for the year field', () => {
        const props = { ...defaultProps, name: 'year' };
        const wrapper = shallow(<Select {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
