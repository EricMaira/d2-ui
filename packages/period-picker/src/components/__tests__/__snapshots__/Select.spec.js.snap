// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select/> Matches the snapshot 1`] = `
<WithStyles(FormControl)
  className="formControl"
>
  <WithStyles(InputLabel)
    htmlFor="test"
  >
    This is a test
  </WithStyles(InputLabel)>
  <WithStyles(Select)
    autoWidth={true}
    inputProps={
      Object {
        "id": "test",
        "name": "test",
      }
    }
    onChange={[MockFunction]}
    value="Daily"
  >
    <WithStyles(MenuItem)
      key="Daily"
      value="Daily"
    >
      Daily
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="Weekly"
      value="Weekly"
    >
      Weekly
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="WeeklyWednesday"
      value="WeeklyWednesday"
    >
      Weekly Wednesday
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="WeeklyThursday"
      value="WeeklyThursday"
    >
      Weekly Thursday
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="WeeklySaturday"
      value="WeeklySaturday"
    >
      Weekly Saturday
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="WeeklySunday"
      value="WeeklySunday"
    >
      Weekly Sunday
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="BiWeekly"
      value="BiWeekly"
    >
      Bi weekly
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="Monthly"
      value="Monthly"
    >
      Monthly
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="BiMonthly"
      value="BiMonthly"
    >
      Bi-monthly
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="Quarterly"
      value="Quarterly"
    >
      Quarterly
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="SixMonthly"
      value="SixMonthly"
    >
      Six monthly
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="SixMonthlyApril"
      value="SixMonthlyApril"
    >
      Six monthly starting in April
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="SixMonthlyNov"
      value="SixMonthlyNov"
    >
      Six monthly starting in November
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="Yearly"
      value="Yearly"
    >
      Yearly
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="FinancialApril"
      value="FinancialApril"
    >
      Financial year starting in April
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="FinancialJuly"
      value="FinancialJuly"
    >
      Financial year starting in July
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="FinancialOct"
      value="FinancialOct"
    >
      Financial year starting in October
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="FinancialNov"
      value="FinancialNov"
    >
      Financial year starting in November
    </WithStyles(MenuItem)>
  </WithStyles(Select)>
</WithStyles(FormControl)>
`;
